CI failing on mypy step

First started [3 days ago](https://github.com/nextstrain/augur/actions/runs/12612501086/job/35149294996):

```
augur/data/__init__.py:25: error: Cannot find implementation or library stub for module named "importlib_resources"  [import-not-found]
augur/data/__init__.py:2[5](https://github.com/nextstrain/augur/actions/runs/12612501086/job/35149294996#step:5:6): note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked [7](https://github.com/nextstrain/augur/actions/runs/12612501086/job/35149294996#step:5:8)9 source files)
Error: Process completed with exit code 1.
```

Caused by change of runner image from ubuntu-22.04 (Python 3.10.12) to ubuntu-24.04 (Python 3.12.3)

## Possible solution

Tell mypy to ignore the conditional import lines. This is already done elsewhere:

https://github.com/nextstrain/augur/blob/f758ced4f7320078eeb4b63f3e2890a175b48617/augur/filter/include_exclude_rules.py#L17-L21

--- a/augur/curate/abbreviate_authors.py
+++ b/augur/curate/abbreviate_authors.py
@@ -52,7 +52,7 @@ def register_parser(
 ) -> argparse._SubParsersAction:
     parser = parent_subparsers.add_parser(
         "abbreviate-authors",
-        parents=[parent_subparsers.shared_parser],  # type: ignore
+        parents=[parent_subparsers.shared_parser],  # type: ignore[attr-defined]
         help=first_line(__doc__),
     )
 

--- a/augur/curate/parse_genbank_location.py
+++ b/augur/curate/parse_genbank_location.py
@@ -50,7 +50,7 @@ def register_parser(
 ) -> argparse._SubParsersAction:
     parser = parent_subparsers.add_parser(
         "parse-genbank-location",
-        parents=[parent_subparsers.shared_parser],  # type: ignore
+        parents=[parent_subparsers.shared_parser],  # type: ignore[attr-defined]
         help=first_line(__doc__),
     )
 

--- a/augur/curate/transform_strain_name.py
+++ b/augur/curate/transform_strain_name.py
@@ -42,7 +42,7 @@ def register_parser(
 ) -> argparse._SubParsersAction:
     parser = parent_subparsers.add_parser(
         "transform-strain-name",
-        parents=[parent_subparsers.shared_parser],  # type: ignore
+        parents=[parent_subparsers.shared_parser],  # type: ignore[attr-defined]
         help=first_line(__doc__),
     )
 

--- a/augur/data/__init__.py
+++ b/augur/data/__init__.py
@@ -20,9 +20,9 @@ Resource files.
 import sys
 
 if sys.version_info >= (3, 11):
-    from importlib.resources import files as _files, as_file as _as_file
+    from importlib.resources import files as _files, as_file as _as_file  # type: ignore[import-not-found]
 else:
-    from importlib_resources import files as _files, as_file as _as_file
+    from importlib_resources import files as _files, as_file as _as_file  # type: ignore[import-not-found]
 
 from pathlib import Path
 from typing import ContextManager

--- a/augur/filter/include_exclude_rules.py
+++ b/augur/filter/include_exclude_rules.py
@@ -16,9 +16,9 @@ from . import constants
 
 try:
     # pandas ≥1.5.0 only
-    PandasUndefinedVariableError = pd.errors.UndefinedVariableError  # type: ignore
+    PandasUndefinedVariableError = pd.errors.UndefinedVariableError  # type: ignore[attr-defined]
 except AttributeError:
-    PandasUndefinedVariableError = pd.core.computation.ops.UndefinedVariableError  # type: ignore
+    PandasUndefinedVariableError = pd.core.computation.ops.UndefinedVariableError  # type: ignore[attr-defined, misc]
 
 
 # The strains to keep as a result of applying a filter function.

--- a/augur/parse.py
+++ b/augur/parse.py
@@ -35,7 +35,7 @@ def fix_dates(d: str, dayfirst: bool = True) -> str:
     try:
         try:
             # pandas <= 2.1
-            from pandas.core.tools.datetimes import parsing  # type: ignore
+            from pandas.core.tools.datetimes import parsing  # type: ignore[attr-defined, import-not-found]
         except ImportError:
             # pandas >= 2.2
             from pandas._libs.tslibs import parsing
@@ -44,7 +44,7 @@ def fix_dates(d: str, dayfirst: bool = True) -> str:
             results = parsing.parse_datetime_string_with_reso(d, dayfirst=dayfirst)
         except AttributeError:
             # pandas 1.x
-            results = parsing.parse_time_string(d, dayfirst=dayfirst)  # type: ignore
+            results = parsing.parse_time_string(d, dayfirst=dayfirst)  # type: ignore[attr-defined]
         if len(results) == 2:
             dto, res = results
         else:

